internal/bminventory/inventory_test.go

```diff
@@ -6546,12 +6576,100 @@ var _ = Describe("infraEnvs", func() {
 				})
 				Expect(reply).Should(BeAssignableToTypeOf(installer.NewRegisterInfraEnvCreated()))
 				actual := reply.(*installer.RegisterInfraEnvCreated)
-				var err error
 				i, err = bm.GetInfraEnvInternal(ctx, installer.GetInfraEnvParams{InfraEnvID: *actual.Payload.ID})
 				Expect(err).ToNot(HaveOccurred())
 				err = db.Model(&common.InfraEnv{}).Where("id = ?", i.ID).Update("generated_at", strfmt.DateTime(time.Now().AddDate(0, 0, -1))).Error
 				Expect(err).ToNot(HaveOccurred())
 			})
+			jsonEncodeStringArray := func(array []string) string {
+				b, e := json.Marshal(&array)
+				Expect(e).ToNot(HaveOccurred())
+				return string(b)
+			}
+			Context("Update DiscoveryKernelArguments", func() {
+				ka1 := models.KernelArguments{
+					"p1",
+					"p2",
+				}
+				ka2 := models.KernelArguments{
+					"p3",
+					"p4",
+					"p1",
+					"p5",
+				}
+				tests := []struct {
+					name                    string
+					initialKernelArguments  models.KernelArguments
+					updateKernelArguments   models.KernelArguments
+					expectedKernelArguments models.KernelArguments
+				}{
+					{
+						name: "verify discovery kernel arguments are not set when none are set in update",
+					},
+					{
+						name:                    "update discovery kernel arguments when none were previously set",
+						updateKernelArguments:   ka1,
+						expectedKernelArguments: ka1,
+					},
+					{
+						name:                    "verify existing discovery kernel arguments are not set when none are set in update",
+						initialKernelArguments:  ka1,
+						expectedKernelArguments: ka1,
+					},
+					{
+						name:                    "overwrites existing kernel arguments with new ones",
+						initialKernelArguments:  ka1,
+						updateKernelArguments:   ka2,
+						expectedKernelArguments: ka2,
+					},
+					{
+						name:                   "clears the discovery kernel args when updated to an empty array",
+						initialKernelArguments: ka2,
+						updateKernelArguments:  models.KernelArguments{},
+					},
+				}
+				for index := range tests {
+					t := tests[index]
+					It(t.name, func() {
+						Expect(i.DiscoveryKernelArguments).To(BeNil())
+						var reply middleware.Responder
+						if t.initialKernelArguments != nil {
+							mockInfraEnvUpdateSuccess()
+							reply = bm.UpdateInfraEnv(ctx, installer.UpdateInfraEnvParams{
```

Comment:
I would prefer setting this using the database for the precondition.

Reply:
I don't agree.  I prefer to set it with API.

Reply:
Okay, pretty much every other infraenv test uses the DB to initialize the test state, but if you feel strongly I guess this one can do setup over the API :man_shrugging: